lib/buildkite/test_collector/library_hooks/rspec.rb

```diff
@@ -28,9 +29,36 @@
 
     trace = Buildkite::TestCollector::RSpecPlugin::Trace.new(example, history: tracer.history)
     Buildkite::TestCollector.uploader.traces[example.id] = trace
+
+    Thread.current[:_buildkite_send_queue].push(trace)
+
+    data = if Thread.current[:_buildkite_send_queue].size > Buildkite::TestCollector.batch_size
+      Thread.current[:_buildkite_send_queue].shift(Buildkite::TestCollector.batch_size)
+    end
+
+    if data
+      new_thread = Buildkite::TestCollector::Uploader.upload(data)
+      Thread.current[:_buildkite_upload_threads] << new_thread if new_thread
+    end
   end
 
   config.after(:suite) do
+    puts "🤡🤡🤡 okay tests have finished"
+    data = Thread.current[:_buildkite_send_queue].size.times.collect { Thread.current[:_buildkite_send_queue].pop }
+
+    if data
+      new_thread = Buildkite::TestCollector::Uploader.upload(data)
+      Thread.current[:_buildkite_upload_threads] << new_thread if new_thread
+    end
+
+    puts "🫡🫡🫡 waiting up 60s for threads to finish #{Time.now}"
+    Thread.new(Thread.current[:_buildkite_upload_threads]) do |threads|
+      threads.each { |t| t.join(60) }
+    end.join(60)
+    puts "🫡🫡🫡 have finished waiting #{Time.now}"
+
+    Thread.current[:_buildkite_upload_threads].each { |t| t&.kill }
```

Comment:
Safely terminate any threads that haven't finished sending data via the upload api